### DISCUSSION TITLE
Allow custom comparison for cache invalidation

### DISCUSSION THREAD

**Discussion by BickelLukas** [SixLabors/ImageSharp.Web#329] [Ideas]
It would be great if we could provide a custom delegate in `IImageResolver` that allows us to compare the cache and the image to find out whether the cache is still valid. 

Right now the comparison is done based on the lastWriteTimeUtc parameter of the ImageMetadata. This does not work if the image storage does not save the write time for files. 

I am storing images in ravendb attachments. These attachments do not have a lastWriteTime, but they provide a changeVector that could be used for comparison. 

Most likely this would also need to support returning and storing custom information for `ImageMetadata`

**Comment by JimBobSquarePants** [comment]
Wouldn't the last write time be on the document itself? The docs suggest that.

https://ravendb.net/docs/article-page/5.4/csharp/document-extensions/attachments/what-are-attachments#example-ii

**Reply by BickelLukas** [reply]
The last write time on the document doesn't correspond to the one of the attachment. In our system a document can be modified without modifying attachments and also a document can have multiple attachments. 

Using the last modified of the document will improve the caching but it still wouldn't be ideal. For that we need to compare the changevector or the hash of the attachment itself.
